wdl-analysis/src/analyzer.rs

```diff
@@ -447,37 +500,83 @@ impl Analyzer {
     /// the analyzer, those documents will be removed.
     ///
     /// Documents are only removed when not referenced from importing documents.
-    pub async fn remove_documents<'a>(&self, uris: Vec<Url>) {
+    pub async fn remove_documents<'a>(&self, documents: Vec<Url>) {
```

Comment:
This should probably return a `Result<()>` same as adding documents.

Reply:
What are you thinking would be the error case? It's quite intentional that we ignore documents that weren't previously added as this gets called from a file change notification in the LSP, which may be providing us directories that don't contain any WDL documents.

Reply:
I think this is wrapped up in the last question of "should this return an error if the queue fails to acknowledge the removal, thus indicating some issue with the queue".

Reply:
The `Analyzer` methods now all return `Result`.